epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyVisibilityTracker.java

```diff
@@ -58,10 +65,46 @@
 
   private boolean onChangedEnabled = true;
 
+  /** All nested visibility trackers */
+  private Map<RecyclerView, EpoxyVisibilityTracker> nestedTrackers = new HashMap<>();
+
   /** This flag is for optimizing the process on detach. If detach is from data changed then it
    * need to re-process all views, else no need (ex: scroll). */
   private boolean visibleDataChanged = false;
 
+  /**
+   * Register a nested recycler view. If the parent is a recycler view with a visibility tracker
+   * attached the visibility tracking will be enabled on the nested. The parent will also notify
+   * the nested tracker if necessary.
+   *
+   * You are responsible to call this method if you are not using
+   * {@link com.airbnb.epoxy.EpoxyRecyclerView}
+   *
+   * @param nestedRecyclerView the nested recycler view
+   * @return a visibility tracker if one was attached
+   */
+  @Nullable
+  public static EpoxyVisibilityTracker registerNestedRecyclerView(
+      @NonNull RecyclerView parent,
+      @NonNull RecyclerView nestedRecyclerView
+  ) {
+    EpoxyVisibilityTracker nestedTracker = ATTACHED_TRACKERS.get(nestedRecyclerView);
+    if (nestedRecyclerView.getAdapter() instanceof BaseEpoxyAdapter) {
+      // Only attach a tracker if the nested RecyclerView use Epoxy and don't have a tracker yet
+      nestedTracker = ATTACHED_TRACKERS.get(nestedRecyclerView);
+      if (nestedTracker == null) {
+        EpoxyVisibilityTracker parentTracker = ATTACHED_TRACKERS.get(parent);
```

Comment:
this system of using static references to check attached states seems flimsy.

I see the need, but possibly have a better idea. What if we used view tags to store the tracker for a recyclerview instead? So when you call `attach` the attacher adds itself as a view tag associated with a custom id, and then for any other recyclerview in the future we can get it's attacher and know if it is set up for visibiility tracking or not?

Reply:
I like the idea of using tag better ! Let me check that...